src/main/scala/tile/BaseTile.scala

```diff
@@ -172,16 +172,26 @@ abstract class BaseTile private (val crossing: ClockCrossingType, q: Parameters)
   protected val tlSlaveXbar = LazyModule(new TLXbar)
   protected val intXbar = LazyModule(new IntXbar)
 
+  // Node for legacy instruction trace from core
   val traceSourceNode = BundleBridgeSource(() => Vec(tileParams.core.retireWidth, new TracedInstruction()))
   val traceNode = BundleBroadcast[Vec[TracedInstruction]](Some("trace"))
   traceNode := traceSourceNode
 
+  // Trace sideband signals into core
   val traceAuxNode = BundleBridgeNexus[TraceAux]()
   val traceAuxSinkNode = BundleBridgeSink[TraceAux]()
   val traceAuxDefaultNode = BundleBridgeSource(() => new TraceAux)
   traceAuxSinkNode := traceAuxNode := traceAuxDefaultNode
```

Comment:
I know this is older code, but why do we make a `BundleBridgeSink` and `BundleBridgeSource` in the same level of hierarchy? Why not just have a single `BundleBroadcast` and get rid of the `BBSink`/``BBSource` nodes?

Reply:
Let's start by saying I don't have a thorough understanding of BundleBridge.  The thinking at the time this was written was that both the source and sink of TraceAux were optional.  There can be zero or one sinks (cores) and zero or more sources (trace clients).  BundleBroadcast requires exactly one source so does not work in this situation.

The RocketTile level connects I/O signals from the core directly to the traceSourceNode and may or may not connect I/O signals to traceAuxSinkNode.
- If there is a TraceAux sink but no source, such as when there is no trace client in the system, the default source ends up driving the sink.
- If there is a trace client source but the core doesn't connect to the sink, that's also ok.  
- If there is a sink and one or more sources, then the connection is made using the TraceAux logic in BaseTileModuleImp (the logical OR described in the submission).